agent/xds/rbac.go

```diff
@@ -205,18 +216,47 @@ func removePermissionPrecedence(perms []*rbacPermission, intentionDefaultAction
 		}
 
 		perm.ComputedPermission = perm.Flatten()
+
+		//attach jwt permissions
+		perm.ComputedPermission = perm.GenerateJWTPermissions()
 		out = append(out, perm)
 	}
 
 	return out
 }
 
+func (p *rbacPermission) GenerateJWTPermissions() *envoy_rbac_v3.Permission {
+	if p.jwtInfos == nil {
+		return p.ComputedPermission
+	}
+
+	var jwtPerms []*envoy_rbac_v3.Permission
+
+	for _, info := range p.jwtInfos {
+		claimsPermission := jwtInfosToPermission(info.ProviderRequirement.VerifyClaims, info.MetadataPayloadKey)
+		segments := pathToSegments([]string{"iss"}, info.MetadataPayloadKey)
+		issuerPermission := segmentToPermission(segments, info.Provider.Issuer)
+
+		perm := andPermissions([]*envoy_rbac_v3.Permission{
+			issuerPermission, claimsPermission,
+		})
+		jwtPerms = append(jwtPerms, perm)
+	}
+	if len(jwtPerms) > 0 {
```

Comment:
And following from my previous suggestion, we won't need this `if len(jwtPerms) > 0 {` check, if the method is changed to check that jwtPerms is empty at the top, and does an early return.